### ISSUE TITLE
Update the documentation and code with the maximum number of required confirmations

### ISSUE THREAD

**Issue by akolotov** [poanetwork/tokenbridge-contracts#327]
Both methods `hasEnoughValidSignatures` defined in `contracts/libraries/Message.sol` iterate through the set of signatures received from the bridge validators. The number of signatures equals to [the number of signatures required to consider a relay request confirmed](https://github.com/poanetwork/tokenbridge-contracts/blob/20d262702defd3aed39870f3379f1cf0c314305e/contracts/upgradeable_contracts/BaseBridgeValidators.sol#L78). If this number is too big it could lead that "out of gas" error happens during execution of the method `executeSignatures` and the withdrawal request will never be executed.

Consider to do the following:
* perform a research and update the documentation with maximum number of validators which will not lead to the withdrawal request execution interruption;
* update `setRequiredSignatures` and the `initialize` methods in both `RewardableValidators` and `BridgeValidators` with the check that requested value does not exceed the recommended maximum;
* update tests that will confirm that the invocation of `executeSignatures` with the recommended maximum of signatures does not exceed some default gas limit (e.g. gas limit for the Ethereum Mainnet).

**Comment by k1rill-fedoseev** [comment]
I have performed a research similar to the one in #326. Since we have defined max number of validators as `50`, I have tried to use `50` signatures. Here are the results:

In Petersburg, it takes approximately `1442674 - 1529881` gas to run `executeSignatures`, depending on bridge mode. Similarly, in Istanbul, it takes `1480282 - 1547001` gas. 
So, no special limit on number of signatures is needed, since it is already bounded by the maximum number of validators.

For greater sizes of validators list, the actual limit is somewhere near `145` signatures, which is close to limits coming from #326, again, no special limit is needed, limiting validators list size by `140` will be enough.
